lib/spack/spack/cmd/mirror.py

```diff
@@ -97,6 +98,19 @@ def setup_parser(subparser):
         default=spack.cmd.default_list_scope(),
         help="configuration scope to read from")
 
+    # Binary
+    upload_parser = sp.add_parser('binary', help=mirror_binary.__doc__)
+    upload_parser.add_argument('-d', '--directory', default=None,
+                               help="directory containing binaries for mirroring")
+    upload_parser.add_argument('-f', '--file', default=None,
```

Comment:
Same with this one — don’t allow the user to upload any old file. They should upload the archive for a spec, but they shouldn’t have to have it on hand, as spack already knows what files are needed to build a spec.